crates/subspace-farmer-components/Cargo.toml

```diff
@@ -16,6 +16,7 @@ include = [
 bench = false
 
 [dependencies]
+async-lock = "2.8.0"
```

Comment:
This change should probably go at least as a separate commit. Non-critical though.

Reply:
It was here because Clippy was complaining about sync lock held across `.await` (was undetected in previous versions) and to fix it async lock is necessary. Due to small scale of changes I didn't do it in a separate commit.

Reply:
Makes sense then.